source/core/src/main/com/csse3200/game/components/player/PlayerAchievementComponent.java

```diff
@@ -0,0 +1,196 @@
+        }
+        if (energyDrinksCollected.size() == 3) {
+            addAchievement("All three energy drink types collected", "images/items/energy_drink_blue.png");
+        }
+    }
+
+    /**
+     * Handles achievements for when an enemy is defeated.
+     * @param deadCount the amount of animals defeated
+     */
+    public void handleDefeatedEnemyAchievement(int deadCount) {
+
+        switch (deadCount) {
+            case 1:
+                addAchievement("First defeated enemy", "images/items/energy_drink_blue.png");
+                break;
+            case 10:
+                addAchievement("10 enemies defeated", "images/items/energy_drink_blue.png");
+                break;
+            case 100:
+                addAchievement("100 enemies defeated", "images/items/mystery_box_green.png");
+                break;
+            case 1000:
+                addAchievement("1000 enemies defeated", "images/items/mystery_box_green.png");
+                break;
+            default:
+                // Handle other cases or do nothing
+                break;
+        }
+    }
+
+    /**
+     * Handles achievements for coins collected throught a game
+     */
+    public void handleCoinsAchievement() {
+        int numCoins = entity.getComponent(CoinsComponent.class).getCoins();
+        switch (numCoins) {
+            case 100:
+                addAchievement("Scored 100 coins", "images/items/coin.png");
+                break;
+            case 1000:
+                addAchievement("scored 1000 coins", "images/items/coin.png");
+                break;
+            case 10000:
+                addAchievement("scored 10000 coins", "images/items/coin.png");
+                break;
+            default:
+                break;
+        }
+    }
+
+    /**
+     * Adds the name and the icon of an achievement to a hashMap and
+     * display it on UI
+     *
+     * @param name name of the achievement
+     * @param icon path to image file
+     *
+     */
+    public void addAchievement(String name, String icon) {
+
+        if (!this.achievements.containsKey(name)){
+            //Add achievement to the inner list
+            this.achievements.put(name, icon);
+            // Create a Label to display the achievement message
+            String text = String.format("Congratulations, you have achieved the '%s' achievement!", name);
+            Label achievementLabel = new Label(text, skin, "small");
+//            Table table = new Table();
+//            table.center();
+//            table.setFillParent(true);
+            this.table.row().padTop(5f);
+            this.table.add(achievementLabel).padTop(5f);
+            stage.addActor(table);
+            // unrender the label after 1 second of display
+            Timer.schedule(new Timer.Task() {
+                @Override
+                public void run() {
+                    achievementLabel.remove(); // Remove the label from the screen
+                }
+            }, 3);
```

Comment:
magic 3, could you make this a constant called "ACHIEVEMENT_DURATION" or something of the like

Reply:
resolved and pushed !